docs/DeveloperGuide.adoc

```diff
@@ -364,6 +365,136 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
 
 Certain properties of the application can be controlled (e.g App name, logging level) through the configuration file (default: `config.json`).
 
+=== Command Box Helper
+
+The command box helper is meant to work similarly to the auto fill functions in IDEs. Two different classes are created, `CommandBoxHelper` and `HelperCard`.
+`CommandBoxHelper` contains a ListView which is populated by VBox containers holding the `HelperCard` class. `CommandBoxHelper` retrieves a list of commands and command templates from the `logic` class and performs a check to populate the listview everytime the user enters a key in the command box
+The commands are matched using the `contain()` function and comparing the first character of the user's input. Commands that are matched are placed in a list and mapped to their corresponding `HelperCard` classes.
+For example, an "add" command will create a new instance of `HelperCard` using "add" as a parameter for the `HelperCard` constructor.
+
+To emulate the behaviour of command helpers in IDEs, the use can also navigate the list of commands using the arrow keys. To prevent it from overwriting the existing code of loading input history, an if-else statement is used. The command box helper takes priority in this case; as long as there is a command
```

Comment:
`the use can also` -> `the user can also`

Also, can you illustrate the if-else better with, maybe, an Activity Diagram?